### PR TITLE
[go] Improve examples generation

### PR DESCRIPTION
**Pull request by zippolyte**
Generation of go examples in API docs gives code that doesn't compile: there are a couple typos, issues with types and pointers.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by zippolyte** [comment]
cc @antihax (2017/11) @grokify (2018/07) @kemokemo (2018/09) @bkabrda (2019/07)
Also @jimschubert or @wing328

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java [hunk 1]
```diff
@@ -565,13 +565,6 @@ private String constructExampleCode(CodegenModel codegenModel, HashMap<String, C
             processedModelMap.put(model, 1);
         }
 
-        example = "" + goImportAlias + "." + codegenModel.name + "{";
-        List<String> propertyExamples = new ArrayList<>();
-        for (CodegenProperty codegenProperty : codegenModel.allVars) {
-            propertyExamples.add(codegenProperty.name + ": " + constructExampleCode(codegenProperty, modelMaps, processedModelMap));
-        }
```
Comment (jirikuncar) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java:572]: We should probably include at least required arguments.

Comment (zippolyte) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java:572]: Yeah I wasn't too sure what to do here, keeping simplicity of example, where you'd have to add manually the required arguments, or having everything in a single long line like before, and losing that readability.

I'll try to use the constructor with required arguments to see how it looks

**Comment by jimschubert** [comment]
CircleCI build was a timeout. I've restarted that job.

**Comment by zippolyte** [comment]
Good to merge @jimschubert ?

**Comment by zippolyte** [comment]
Friendly bump on this @jimschubert @wing328 🙇‍♂️

**Comment by wing328** [comment]
@zippolyte can you please resolve the merge conflicts when you've time? I'll try to review later today or tomorrow.

**Pull request merged**